### PR TITLE
22 list item detail view

### PR DESCRIPTION
**Pull request by Nikomus**
https://trello.com/c/M2r0E8Jk/22-list-item-detail-view-vvv-ee

Builds on #18 to add details of each service

### CODE REVIEW

**Comment by marxian** [comment]
#### 📝Changed routes:


| [**/**](https://my-best-life-pwb4nu2my.vercel.app/) | [**/service/\[id\]**](https://my-best-life-pwb4nu2my.vercel.app/service/[id]) |
|:-:|:-:|
|<a href="https://my-best-life-pwb4nu2my.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/vpAo3rZuQHk9ZR_8QK2v_fx56m0_.png">(view full size)</a> |<a href="https://my-best-life-pwb4nu2my.vercel.app/service/[id]">Screenshot of /service/[id]</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/l9LLzjFJ5h8vB3tKKf0kHqLvR8o_.png">(view full size)</a> |







Commit 4a8172d0a28c6f141cb178c24e9db6c711a5c36c (https://my-best-life-pwb4nu2my.vercel.app).

File: .prettierrc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
   "semi": false,
-  "singleQuote": true
+  "singleQuote": true,
+  "endOfLine": "auto"
```
Comment (marxian) [inline .prettierrc:4]: This is good.May I also suggest adding .editorconfig ? https://editorconfig.org/

File: cms/services.ts [hunk 2]
```diff
@@ -18,18 +19,25 @@ interface Service {
   location?: string
 }
 
-export interface ServicePreview {
+export interface ServiceDetail {
```
Comment (marxian) [inline cms/services.ts:22]: Maybe Pick from Service?

https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys

File: pages/index.tsx [hunk 3]
```diff
@@ -1,49 +1,153 @@
...
+import { Service, getServices } from '../cms/services'
+import { CardList } from '../src/Components/CardList'
+import { Layout } from '../src/Components/Layout'
+import { VisuallyHidden } from '../src/Components/VisuallyHidden'
+
+interface ListPageProps {
+  services: Array<ServicePreview>
+}
+
+export interface ServicePreview {
```
Comment (marxian) [inline pages/index.tsx:13]: This too looks like a local Picked interface? I couldn't work out why it was exported from here?

File: src/Components/CardList.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+
+import { Card } from './Card'
+import { ServicePreview } from '../../pages/index'
```
Comment (marxian) [inline src/Components/CardList.tsx:4]: Ah! Here! The export makes sense now. Apols.

**Review by marxian — APPROVED** [review]
Cool.

I think Picking from the maximal type will really help here, since it'll reduce the chance of drift. Assuming I've understood things correctly. I'm still a TypeScript noob

File: cms/services.ts (hunk 2 above)
Comment (Nikomus) [reply cms/services.ts:22]: This is really useful, thank you! `ServicePreview` and `ServiceDetail` are indeed both different subsets of `Service`, I've carded to make them pick in the next card that needs new keys.

**Pull request merged**